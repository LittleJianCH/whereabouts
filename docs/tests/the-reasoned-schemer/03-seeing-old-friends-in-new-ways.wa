import { Null, Cons, Car, Cdr } from "02-teaching-old-toys-new-tricks.wa"

List l
----------- {
  Null l
}

List l
------------- {
  Cdr [l, d]
  List d
}

query x {
  List ["a", "b", x, "d"]
}

query x limit 1 {
  List ["a", "b", ...x]
}

query x limit 3 {
  List ["a", "b", ...x]
}
