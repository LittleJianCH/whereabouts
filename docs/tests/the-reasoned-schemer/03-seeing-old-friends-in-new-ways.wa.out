[ "?x" ]
[ [] ]
[ [], ["?a_82"], ["?a_82", "?a_89"] ]
